### ISSUE TITLE
Any type as a key

### ISSUE THREAD

**Issue by xoac** [jonathonl/goodform#4]
Hi,
object is defined as:
`  typedef std::map<std::string, goodform::variant> object;`

shuldn't be like this?
` typedef std::map<goodform::variant, goodform::variant> object;`

https://github.com/msgpack/msgpack/issues/217

**Comment by jonathonl** [comment]
I would argue that complex keys are out of scope with this library. The original purpose of this library was to validate REST api requests. 

I plan to replace goodform::variant with std::any at some point, and I will consider expanding support for more msgpack types when I get around to doing that conversion. Though I expect that complex keys will not meld well with the library design.

**Comment by jonathonl** [comment]
Fixed with 6780676419217d8ffd9296e3a35ad08bc84bb7bb.

Example:
```
typedef goodform::basic_msgpack<std::vector<std::pair<goodform::any, goodform::any>>> any_key_msgpack;
goodform::any v;
any_key_msgpack::deserialize(input_stream, v);
```
